metrics.go

```diff
@@ -274,24 +336,57 @@ func (m *metrics) doPost(url *url.URL, payload interface{}) (*http.Response, err
 	return m.options.httpClient.Do(req)
 }
 
+func (m *metrics) getOrCreateCounter(name string) *toggleCounters {
+	c, ok := m.counters.Load(name)
+	if ok {
+		return c.(*toggleCounters)
+	}
+
+	nc := &toggleCounters{
+		variants: make(map[string]int64),
+	}
+	actual, _ := m.counters.LoadOrStore(name, nc)
+	return actual.(*toggleCounters)
+}
+
+func (m *metrics) reinsertBucket(bucket api.Bucket) {
```

Comment:
Previously the code just dropped variants on the floor when re-inserting. Needed to get fixed and now seems like a good time I guess so here's a fix